packages/react-native/src/Authenticator/Authenticator.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useMemo } from 'react';
+
+import {
+  AuthenticatorProvider as Provider,
+  AuthenticatorComponentDefaults,
+  AuthenticatorMachineContext,
+  resolveAuthenticatorComponents,
+  useAuthenticator,
+  useAuthenticatorRoute,
+  useAuthenticatorInitMachine,
+  UseAuthenticator,
+} from '@aws-amplify/ui-react-core';
+
+import { AuthenticatorProps, FormHandlers } from './types';
+
+const DEFAULTS = {
+  // TODO: remove cast after adding missing components
+} as unknown as AuthenticatorComponentDefaults<FormHandlers>;
+
+const isAuthenticatedRoute = (route: UseAuthenticator['route']) =>
```

Comment:
Would this be better placed in a separate utils file?

Reply:
If we need it in other places we can abstract it out but since it's such a minimal util think it's fine here for now

Reply:
How would one discover that this util exists here and not duplicate it in another component?

Reply:
How would it living in a _util.ts_ as a sibling to _Authenticator.tsx_ component be any different? 

If we really want to think about whether this will be reused, we are not exposing `route` anywhere else in the UI besides in `Authenticator`, so unless we need to suddenly change the architecture of the functional components, there is really no chance that this util will ever be reused